app/service/rest_svc.py

```diff
@@ -192,6 +195,16 @@ def __init__(self):
             return await self.get_service('data_svc').locate('agents', match=dict(group=group))
         return await self.get_service('data_svc').locate('agents')
 
+    async def add_special_operation_modifier(self, name, func):
+        """
+        Call a special function when a new operation is created
+
+        :param name:
+        :param func:
+        :return:
+        """
+        self.special_operation_modifiers[name] = func
```

Comment:
interesting idea. I like this. I'm wondering if this can be combined with the add_special_payload function on the file_svc. they do the same functionality, just different applications (one adds special functionality on a download, another when an operation starts). seems like they can be consolidated.

Reply:
They definitely could be consolidated. This may just be a result of not wanting change, but it seems to me like the current separation makes sense to me though.
Special payloads require the HTTP request object to be passed, while special initializers require the operation. That differentiation would need to be made when calling the consolidated function, which would either add another parameter + code based on whether we want a payload or an initializer. Or request/operation would need to be passed as a kwarg (I think), which would make following the code less intuitive. Unless there's another, better way to do this...
@privateducky tagging you because I don't know if Github sends notifications for comment replies

Reply:
interesting enough, i'm running into a separate need for this functionality as well. i'm going to put a little thought into it today.